bin/git/getCommitAndBranch.js

```diff
@@ -73,19 +73,20 @@ export async function getCommitAndBranch({ inputFromCI } = {}) {
   // On certain CI systems, a branch is not checked out
   // (instead a detached head is used for the commit).
   if (branch === 'HEAD' || !branch) {
-    ({ branch } = envCi());
-    if (branch === 'HEAD' || !branch) {
-      // $HEAD is for netlify: https://www.netlify.com/docs/continuous-deployment/
-      // $GERRIT_BRANCH is for Gerrit/Jenkins: https://wiki.jenkins.io/display/JENKINS/Gerrit+Trigger
-      // $CI_BRANCH is a general setting that lots of systems use
-      branch =
-        process.env.HEAD ||
-        process.env.GERRIT_BRANCH ||
-        process.env.CI_BRANCH ||
-        process.env.GITHUB_REF ||
-        branch ||
-        'HEAD';
-    }
+    const { prBranch: prBranchFromEnvCi, branch: branchFromEnvCi } = envCi();
+
+    // $HEAD is for netlify: https://www.netlify.com/docs/continuous-deployment/
+    // $GERRIT_BRANCH is for Gerrit/Jenkins: https://wiki.jenkins.io/display/JENKINS/Gerrit+Trigger
+    // $CI_BRANCH is a general setting that lots of systems use
+    branch =
+      prBranchFromEnvCi ||
+      process.env.HEAD ||
+      process.env.GERRIT_BRANCH ||
+      process.env.CI_BRANCH ||
+      process.env.GITHUB_REF ||
+      branchFromEnvCi ||
+      branch ||
```

Comment:
@ndelangen this changes the logic when `prBranch` is undefined (which it is for many CI systems or when there is no PR), as it will now preference the environment vars. I'd prefer not to change the behaviour in such cases.

Reply:
I thought this would be identical, can you explain to me what I need to change?

Reply:
We might just need to be a bit more careful. 

Previously we grabbed `envCi().branch` and used it unless it was falsey or 'HEAD'.

We should probably do something like:

1. Use the git branch unless it is falsey or HEAD (existing check on line 75)
2. Use `envCi().prBranch` unless it is falsey or HEAD
3. Use `envCi().branch` unless it is falsey or HEAD
4. Use various env vars
5. Fall back to 'HEAD' or nothing

Reply:
I think this is resolved now